server/src/main/java/org/opensearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -529,6 +529,21 @@ private static void ensureRemoteStoreNodesCompatibility(DiscoveryNode joiningNod
         }
 
         if (STRICT.equals(remoteStoreCompatibilityMode)) {
+            Optional<DiscoveryNode> existingRemoteIndexStoreNodes = existingNodes.stream()
```

Comment:
@rajiv-kv why do we need change in STRICT mode, this should be put if nodes are in mixed mode?

Reply:
MIXED mode has additional validations relevant to the remote index store (data replication)
 - nodes of higher versions are not allowed to join the cluster [version upgrade is blocked]
 - dependent setting `cluster.migration.direction`  for data migration.
 
It looks to me that coupling with MIXED mode might bring in un-necessary dependencies with data migration and break else-where.